corehq/apps/locations/permissions.py

```diff
@@ -134,7 +134,9 @@ def is_locations_admin(view_fn):
 
 
 def user_can_edit_any_location(user, project):
-    return user.is_domain_admin(project.name) or not project.location_restriction_for_users
+    return user.is_domain_admin(project.name) or \
+        not project.location_restriction_for_users or \
+        user.has_permission(project.name, 'access_all_locations')
```

Comment:
This will break the `location_restriction_for_users` model of permissions - those projects pre-date the `access_all_locations` permission, so all their user roles are defaulted to have that permission.